### ISSUE TITLE
Add the \Scripts directory to the PATH when installing Python

### ISSUE THREAD

**Issue by ryanewtaylor** [chocolatey/chocolatey-coreteampackages#111]
When installing Python via Chocolatey I noticed that I was unable to use PIP from the command line as the python \Scripts directory was not added to the environment path. PIP is included in newer versions of Python and so it would be helpful to have this available on the PATH as well. 

To confirm the behavior of the MSI I ran it in GUI mode and via msiexec. The default behavior of both is to not add the python.exe and scripts dir to PATH but there is an option to do so in the MSI itself. You can do this through msiexec by adding `ADDLOCAL=ALL`. as an argument.

If this is done, perhaps Chocolatey would not need to add it to the path itself?

**Comment by ferventcoder** [comment]
I agree this should be done.

**Comment by ryanewtaylor** [comment]
If the maintainers do not have time for this addition, Is this something that I could potentially do via a pull request? I think the solution would be to:

1, Sign the CLA
2. Fork this repository
3. Create a branch such as `add-pythons-scripts-dir-to-path`
4. Add `ADDLOCAL=ALL` to `$partialInstallArgs = '/qn /norestart ALLUSERS=1 TARGETDIR='` as an argument in the chocolateyInstall.ps1 file
5. Remove `Install-ChocolateyPath $installPath 'Machine'` from the chocolateyInstall.ps1 file
6. Repeat for each python2 package (I would need to check what the behavior for the python3 installers are and see if they have similar requirements)
7. Submit a PR back to this repository

I read the contributors page. However I would want to be 100% sure of proper procedure so that I do not disrupt or offend any contributors and users. Please add or correct anything else you think is important and/or required for someone to make this sort of change.

**Comment by ferventcoder** [comment]
I do not believe there is a CLA for the coreteam packages.

**Comment by gep13** [comment]
@ferventcoder correct, there is no CLA for this repo.

@ryanewtaylor would be happy to accept a PR to get this change included.

**Comment by obilodeau** [comment]
For Python3 this is done using `PrependPath=1` installer option: https://docs.python.org/3/using/windows.html

**Comment by obilodeau** [comment]
Python 2's problem was likely fixed with: ce24b5c3bc and 5e214d3dd9 which introduced `\Scripts` injection into $env:PATH.

I'm doing a pull request for python 3. I tested it and it works (adding pip installed scripts to path). I can also add `ADDLOCAL=ALL` to Python 2 while I'm at it. Just let me know if interested.

**Comment by ferventcoder** [comment]
> Python 2's problem was likely fixed with: ce24b5c and 5e214d3 which introduced \Scripts injection into $env:PATH.

Those two look like local changes that would not be there once the script exits. In other words, those are not permanent changes to the PATH, only temporary. If this is not what you are looking for, then I would suggest employing an additional fix.

**Comment by obilodeau** [comment]
Ok, I'll update my PR to include the Python 2 fix as well then. Should I
remove the 'machine' thing (it was previously stated to remove it) and path
modifications as well you think?

**Comment by ferventcoder** [comment]
> Ok, I'll update my PR to include the Python 2 fix as well then. Should I
remove the 'machine' thing (it was previously stated to remove it) and path
modifications as well you think?

You may need to clarify by pointing to some code here.

**Comment by obilodeau** [comment]
> You may need to clarify by pointing to some code here.

Sure. I was on mobile yesterday when I wrote this, otherwise I would have included it ;)

I'm referring to [these two lines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/automatic/python2-x86_32/tools/chocolateyInstall.ps1#L78):

```powershell
    Install-ChocolateyPath $installPath 'Machine'
    $env:Path = "$($env:Path);$installPath;$installPath\Scripts"
```

The path was added by the commits (ce24b5c and 5e214d3) and based on the commit message I would say it was a failed attempt at making it permanent and, as you mentioned, these are not permanent changes so I would remove it. Also I suggest to remove the 'Machine' line because of @ryanewtaylor's comment:

> 5. Remove Install-ChocolateyPath $installPath 'Machine' from the chocolateyInstall.ps1 file

**Comment by obilodeau** [comment]
This bug can be closed since #261 was merged.

**Comment by gep13** [comment]
@obilodeau thanks for reminding me :smile:
